### ISSUE TITLE
[editor] support splitting context menu items for editors in the main area

### ISSUE THREAD

**Issue by vince-fugnitto** [eclipse-theia/theia#6665]
**Description**

I propose we add the following context-menu items to **editors** present in the **main area**, to allow users to quickly split editors:

- `split up` 
- `split down`
- `split left`
- `split right`

In the future we can also assign keybindings to these actions.
For reference, they are supported in VS Code:

<img width="327" alt="Screen Shot 2019-11-29 at 10 59 48 AM" src="https://user-images.githubusercontent.com/40359487/69880203-63167800-1297-11ea-91d1-bb065123f284.png">

**Comment by akosyakov** [comment]
It is fine, but what you will show? Does it depend on https://github.com/eclipse-theia/theia/issues/3857?

**Comment by vince-fugnitto** [comment]
Yes I do think it depends on #3857, and I imagined it only for `editors` like in VS Code.

**Comment by vince-fugnitto** [comment]
> It is fine, but what you will show? Does it depend on #3857? What about other widgets?

@akosyakov I also had in mind to support moving editors through a context-menu and keybinding. Instead of having to drag & drop, it'd split and move the editor. Do you think it's a worthwhile feature to support? I believe versions of Jetbrains IDE have the same functionality.

**Comment by akosyakov** [comment]
It would be fine with me, but you are better ask others. Generally being able to do everything with the keyboard is good. Maybe open another issue and ping other committers?

**Comment by vince-fugnitto** [comment]
Closing as the commands are present :+1: 

![image](https://user-images.githubusercontent.com/40359487/223216105-572a507c-d6ec-4696-9423-494c0b89e5a6.png)
